test/services/find_transaction_by_client_id.test.js

```diff
@@ -0,0 +1,48 @@
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+const { describe, it, before, beforeEach } = exports.lab = Lab.script()
+const { expect } = Code
+const Regime = require('../../app/models/regime')
+const { addBillRunTransaction, cleanBillRuns } = require('../helpers/bill_run_helper')
+const CreateBillRun = require('../../app/services/create_bill_run')
+const FindTransactionByClientId = require('../../app/services/find_transaction_by_client_id')
+const Schema = require('../../app/schema/pre_sroc')
+
+describe('Find existing transaction', () => {
+  let regime
+  let schema
+  let region
+  let clientId
+
+  before(async () => {
+    regime = await Regime.find('wrls')
+    schema = Schema[regime.slug]
+    region = 'A'
+    clientId = 'abc123'
+  })
+
+  beforeEach(async () => {
+    await cleanBillRuns()
+  })
+
+  it('returns the transaction id if a transaction with the client id exists', async () => {
```

Comment:
Slight tweak to the wording to be more verbose about

- what gets returned from the service
- what we are actually matching against

```suggestion
  it('returns the matching transaction if one with the same regime and client id exists', async () => {
```